checkov/kubernetes/checks/resource/base_rbac_check.py

```diff
@@ -0,0 +1,86 @@
+from checkov.kubernetes.checks.resource.base_spec_check import BaseK8Check
+from checkov.common.models.enums import CheckCategories, CheckResult
+from typing import Dict, Any, List
+
+
+class RbacOperation():
+    """
+    A collection of RBAC permissions that permit a certain operation within Kubernetes.
+    For example, the RbacOperation below denotes a write operation on admission webhooks.
+    control_webhooks = RbacOperation(
+        apigroups=["admissionregistration.k8s.io"],
+        verbs=["create", "update", "patch"],
+        resources=["mutatingwebhookconfigurations", "validatingwebhookconfigurations"]) 
+    Rules matching an apiGroup, verb and resource should be able to perform the operation.
+    """
+    def __init__(self, apigroups: List[str], verbs: List[str],
+                 resources: List[str]):
+        self.apigroups = apigroups
+        self.verbs = verbs
+        self.resources = resources
+
+
+class BaseRbacK8sCheck(BaseK8Check):
+    """
+    Base class for checks that evaluate RBAC permissions in Roles and ClusterRoles
+    """
+    def __init__(self, name, id, supported_entities=None):
+        if supported_entities is None:
+            supported_entities = ["Role", "ClusterRole"]
+        categories = [CheckCategories.KUBERNETES]
+        super().__init__(name=name, id=id, categories=categories, supported_entities=supported_entities)
+        # A role that grants *ALL* the RbacOperation in failing_operations fails this check
+        self.failing_operations: RbacOperation = []
+
+    def scan_spec_conf(self, conf):
+        if isinstance(conf.get("rules"), list) and len(conf.get("rules")) > 0:
+            role_can_do_operation_arr = [False] * len(self.failing_operations)
+            # Go over all rules
+            for rule in conf["rules"]:
+                # Check if the rule matches an RbacOperation this check looks for
+                for i in range(len(self.failing_operations)):
+                    if not role_can_do_operation_arr[i]:
+                        # RbacOperation wasn't found yet, check rule
+                        role_can_do_operation_arr[i] = self.rule_can(rule, self.failing_operations[i])
+                # If all operations were found, role / clusterRole failed the check
+                if False not in role_can_do_operation_arr:
+                    return CheckResult.FAILED
+        return CheckResult.PASSED
```

Comment:
```suggestion
    def scan_spec_conf(self, conf):
        rules = conf.get("rules")
        if rules and isinstance(rules, list):
            for operation in self.failing_operations:
                # if one operation can't be found, check passes
                if not any(self.rule_can(rule, operation) for rule in rules):
                    return CheckResult.PASSED
            # all operations were found, therefore the check fails
            return CheckResult.FAILED
        
        return CheckResult.PASSED
```
I hope it's ok. I rewrote it a bit 🙂

Reply:
Thanks @gruebel ! looks much better